src/lesson12/Library.java

```diff
@@ -0,0 +1,82 @@
+package lesson12;
+
+import java.util.*;
+
+public class Library {
+    private Map<String, List<Book>> list = new HashMap<>();
```

Comment:
Hou about `Map<String, Map<String, Book>>`?
I think, it will simplify a lot of logic